src/core/system.c

```diff
@@ -1,6 +1,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <time.h>
```

Comment:
is it really needed? there is #include <sys/time.h>
strange... interesting what is GCC version?